logr/fields_test.go

```diff
@@ -0,0 +1,19 @@
+package logr
+
+import (
+	"testing"
+
+	"github.com/sourcegraph/log"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestToLogFields(t *testing.T) {
+	fields := toLogFields([]any{
+		"hello", "world",
+		"goodbye", "bob",
```

Comment:
nit: perphaps add a test where the value is not a string (which makes sense from the test perspective, though we know this is going into `zap.Any` so it'll be ok)

Reply:
Actually, isn't this a case where we could use the type constraints 🤔 ?

Reply:
Added a test with a non-string type! I'm not sure what you mean with constraints though - since the caller uses `any`, I don't think we can offer a more specific type

Reply:
Nevermind, I forgot that type constraints are not union and that wouldn't work. Late evening ideas wasn't completely processed 😆